addons/payment/wizards/payment_link_wizard_views.xml

```diff
@@ -18,6 +19,9 @@
                 <group>
                     <field name="link" readonly="1" widget="CopyClipboardChar"/>
                 </group>
+                <group attrs="{'invisible':[('partner_email', '!=', False)]}">
```

Comment:
```suggestion
                <group attrs="{'invisible':['|', ('partner_id', '=', False), ('partner_email', '!=', False)]}">
```

Reply:
@sswapnesh I think this alert should be shown every time the email is not set. If `partner_id` is False, then there is no email. Which prevent some payment acquirer to work as expected.

Reply:
having partner_id false is basically impossible anyway, since the default_get sets it for both SO and invoice

Reply:
@dbo-odoo Actually, I achieved to create an invoice without a partner. In this case, the message is displayed.